tests/test_citation_counts.py

```diff
@@ -0,0 +1,177 @@
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import os
+
+import time
+
+from collections import Counter
+
+from inspirehep.dojson.processors import convert_marcxml
+
+from inspirehep.modules.citations.tasks import update_citation_count_for_records
+
+from invenio.testsuite import InvenioTestCase, make_test_suite, run_test_suite
+
+from invenio_ext.sqlalchemy import db
+
+from invenio_search.api import Query
+
+import pkg_resources
+
+
+class CitationTestsOnRecordInsert(InvenioTestCase):
+
+    def setUp(self):
+
+        from invenio_records.api import create_record
+        from invenio_records.models import Record
+        self.marcxml_file = pkg_resources.resource_stream('tests',
+                                                          os.path.join(
+                                                              'fixtures',
+                                                              'test_citation_counts.xml')
+                                                          )
+
+        self.converted_record = convert_marcxml(self.marcxml_file).next()
+        self._transaction = db.session.begin_nested()
+        self.cited_recids = [r['recid'] for r in self.converted_record['references'] if r.get('recid')]
+        self.new_recid = self.converted_record['control_number']
+        self.expected_citation_counts_after_insert = Counter()
+        self.citation_counts_after_insert = Counter()
+
+        for cited_recid in self.cited_recids:
+            references_to_cited_recid = Query('refersto:' + str(cited_recid)).search().recids
+            queried_records = Query('control_number:' + str(cited_recid)).search().records()
+            cited_record = queried_records[0] if queried_records else None
+
+            if cited_record:
+                if int(self.new_recid) in references_to_cited_recid:
+                    self.expected_citation_counts_after_insert[cited_recid] = cited_record.get('citation_count', 0)
+                else:
+                    self.expected_citation_counts_after_insert[cited_recid] = \
+                        cited_record.get('citation_count', 0) + 1
+
+        self.record = Record(id=self.new_recid)
+        db.session.add(self.record)
+        create_record(self.converted_record)
+
+        # we need to make sure the new record has been already indexed
+        timeout = 15  # [seconds]
+        timeout_start = time.time()
+        while time.time() < timeout_start + timeout:
+            if len(Query('control_number:' + self.new_recid).search().recids) > 0:
```

Comment:
Here avoid using `.recids` (as per above). Just 
```python
len(Query('control_number:' + self.new_recid).search())
```

will give you the correct number